sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -101,6 +101,10 @@
 # Timeout for a BQ streaming insert RPC. Set to a maximum of 2 minutes.
 BQ_STREAMING_INSERT_TIMEOUT_SEC = 120
 
+_PROJECT_PATTERN = r'([a-z0-9.-]+:)?[a-z][a-z0-9-]{4,28}[a-z0-9]'
+_DATASET_PATTERN = r'\w{1,1024}'
+_TABLE_PATTERN = r'[\p{L}\p{M}\p{N}\p{Pc}\p{Pd}\p{Zs}$]{1,1024}'
```

Comment:
Thanks for the above explanation. Could you  please explain the regex for `_TABLE_PATTERN` ?

Reply:
> Contain Unicode characters in category L (letter), M (mark), N (number), Pc (connector, including underscore), Pd (dash), Zs (space). 

from https://cloud.google.com/bigquery/docs/tables#table_naming

The `$` is to also add support for table partitions.

Reply:
Would you mind adding this link or the comment here in a follow-up PR? Thank you.